subsys/bluetooth/host/gatt.c

```diff
@@ -128,46 +130,95 @@ static int gatt_register(struct bt_gatt_service *svc)
 	return 0;
 }
 
-void bt_gatt_init(void)
-{
-	/* Register mandatory services */
-	gatt_register(&gap_svc);
-	gatt_register(&gatt_svc);
-}
-
-static struct k_sem sc_sem = K_SEM_INITIALIZER(sc_sem, 1, 1);
+static struct gatt_sc {
+	struct bt_gatt_indicate_params params;
+	u16_t start;
+	u16_t end;
+	struct k_sem sem;
+	struct k_delayed_work work;
+} gatt_sc;
 
 static void sc_indicate_rsp(struct bt_conn *conn,
 			    const struct bt_gatt_attr *attr, u8_t err)
 {
 	BT_DBG("err 0x%02x", err);
 
-	k_sem_give(&sc_sem);
+	k_sem_give(&gatt_sc.sem);
```

Comment:
Instead of giving a semaphore here, I think it'd make sense to have a flag "services changed after last SC indication was sent" and if the flag is set resend another indication from the response callback.